### DISCUSSION TITLE
setHeaderRoute to stable - instead of canary

### DISCUSSION THREAD

**Discussion by Idokah** [argoproj/argo-rollouts#3710] [Q&A]
there's an option to add header routing rule that will override the weights and route traffic to canary based on header.
I want to override weights - and route traffic to **stable** based on headers.
is it possible?
thanks

**Comment by rossanthony** [comment]
@Idokah - I don't suppose you managed to solve this? I'm looking to do something similar to solve the problem of rolling out a clientside application with a BFF in a mono-repo. Would be nice to be able to have users sessions remain sticky, i.e. if they have the clientside SPA loaded prior to the rollout starting, it would send a "stable" header, but new users loading the app for the first time during the rollout would get a chance of loading the canary and would then send the "canary" header to the BFF to ensure the two remain in sync for the duration of the roll out.

**Comment by kostis-codefresh** [comment]
Depends on what is your traffic provider.

You should be able to create an httproute/service/whatever that always connects to the stable service when a specific header is present. This does not need to be managed with Argo Rollouts
